.github/workflows/pull-request-ui.yml

```diff
@@ -30,33 +78,69 @@ jobs:
             osFolder: "windows"
 
     steps:
+      - name: Configure Change Filter
+        id: filter
+        shell: bash
+        run: |
+          FILTER_ANY_CHANGE="false"  # default value
+          if [[ "${{ needs.changes.outputs.github }}" == "true" || "${{ needs.changes.outputs.bindings }}" == "true" || "${{ needs.changes.outputs.root }}" == "true" || "${{ needs.changes.outputs.scripts }}" == "true" || "${{ needs.changes.outputs.source }}" == "source" ]]; then
+            FILTER_ANY_CHANGE="true"
+          fi
+
+          echo "any=$FILTER_ANY_CHANGE" >> $GITHUB_OUTPUT
+          echo "github=${{ needs.changes.outputs.github }}" >> $GITHUB_OUTPUT
+          echo "bindings=${{ needs.changes.outputs.bindings }}" >> $GITHUB_OUTPUT
+          echo "root=${{ needs.changes.outputs.root }}" >> $GITHUB_OUTPUT
+          echo "scripts=${{ needs.changes.outputs.scripts }}" >> $GITHUB_OUTPUT
+          echo "source=${{ needs.changes.outputs.source }}" >> $GITHUB_OUTPUT
+
+      # Checkout
       - name: Checkout Code
+        if: (steps.filter.outputs.any == 'true')
```

Comment:
then each step gets an if statement describing the conditions where it should run